### PR TITLE
SI-8675 Avoid unreported error after second try using implicit view

### PR DESCRIPTION
**Pull request by retronym**
This is specific to situations in which we first typecheck an
application `qual.m(arg)` against the method `m` directly provided
by `qual`, and then fall back to `implicitView(qual).m(arg)`.

Regressed in SI-3971 / 7fa77af, in which error reports were moved
to the innermost `Apply`, and the check for `errorInResult` was
accordingly changed to recurse through `Apply` trees.

Before that change, we did not fall back to using a view. After the
change, we do try a view. We retypecheck the arguments under the
`retyping` mode (see `tryTypedArgs`), but this doesn't seem to
be enough to avoid leaking the error typed nested trees from the
first try.

Here's an example from the enclosed test case:

```
a.update(0, x[A]({new isString(true)}))
                               `-- error typed

refArrayOps(a).update(0, x[A]({new isString(true)}))
                         `                  `-- error type persists
                          `-- this tree is retypecked by tryTypedArgs
```

This commit changes `onError` to only proceed with the second
try if the retyped argument trees are error free.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 1]
```diff
@@ -4399,7 +4399,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           if (retry) {
             val Select(qual, name) = fun
             tryTypedArgs(args, forArgMode(fun, mode)) match {
-              case Some(args1) =>
+              case Some(args1) if args1.forall(arg => !arg.exists(_.isErroneous)) =>
```
Comment (adriaanm) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: Can't resist nitpicking: pulling out the negation makes this more succinct.
`if !args1.exists(_.exists(_.isErroneous))`, which makes you wonder whether we should have a closure-free way of asking a tree `_.exists(_.isErroneous)` -- `_.containsError`? It's probably not a hotspot, so consider this thinking out loud.

**Comment by adriaanm** [comment]
Whoops! Probably safe to go into 2.11.2, since it seems hard to exploit (though, given the bug report, not impossible).

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala (hunk 1 above)
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: My dog Roemer has our first fleas in a dozen years -- I didn't act fast enough with the flea stuff -- and now I've become too good at picking fleas off with tweezers.  It's a bit like spear fishing.  But I have to take my glasses off to focus correctly.  Anyway, my point is, you get good at picking nits for a reason.

**Comment by retronym** [comment]
I've attached a flea collar to this patch.

**Comment by adriaanm** [comment]
Good, I'll shelve the shampoo then.

**Comment by retronym** [comment]
@adriaanm I've added another test case based on a report from the mailing list.

**Pull request merged**